packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -136,6 +143,48 @@ export class SelectedNetworkController extends BaseController<
     });
     this.#registerMessageHandlers();
 
+    this.messagingSystem
+      .call('PermissionController:getSubjectNames')
+      .filter(
+        (domain) =>
+          this.state.domains[domain] === undefined &&
+          this.state.domains[domain] === undefined,
+      )
+      .forEach((domain) =>
+        this.setNetworkClientIdForDomain(
+          domain,
+          this.messagingSystem.call('NetworkController:getState')
+            .selectedNetworkClientId,
+        ),
+      );
+
+    this.messagingSystem.subscribe(
+      'PermissionController:stateChange',
+      (_, patches) => {
+        patches.forEach(({ op, path }) => {
+          const isChangingSubject =
+            path[0] === 'subjects' && path[1] !== undefined;
+          if (isChangingSubject && typeof path[1] === 'string') {
+            const domain = path[1];
+            if (op === 'add' && this.state.domains[domain] === undefined) {
+              this.setNetworkClientIdForDomain(
+                domain,
+                this.messagingSystem.call('NetworkController:getState')
+                  .selectedNetworkClientId,
+              );
+            } else if (
+              op === 'remove' &&
+              this.state.domains[domain] !== undefined
+            ) {
+              this.update(({ domains }) => {
+                delete domains[domain];
```

Comment:
prolly better to collect all the changes and do to them in 1 update.

Reply:
I agree with this. Do you still intend to do this as part of this PR? Or in a separate PR?

Reply:
I think we should do it as a seperate PR, since its impact should be pretty low, and because theres actually quite a few places where this can be optimized (the initial loop over permissions & calling setNetworkClientIdForDomain will end up causing multiple state updates as well). 

IMO Before spending more time on it, lets just keep it in mind if we see any perf issues pop up